nutanix/resource_nutanix_ndb_time_machine_cluster.go

```diff
@@ -0,0 +1,232 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if err = d.Set("log_drive_id", resp.LogDrive); err != nil {
+		return diag.Errorf("error setting log_drive_id for tms clusters %s: %s", d.Id(), err)
+	}
+
+	if err = d.Set("log_drive_status", resp.LogDriveStatus); err != nil {
+		return diag.Errorf("error setting description for tms clusters %s: %s", d.Id(), err)
+	}
+
+	if err = d.Set("type", resp.Type); err != nil {
+		return diag.Errorf("error setting name for tms clusters %s: %s", d.Id(), err)
+	}
+
+	if err = d.Set("description", resp.Description); err != nil {
+		return diag.Errorf("error setting description for tms clusters %s: %s", d.Id(), err)
+	}
+
+	if err = d.Set("status", resp.Status); err != nil {
+		return diag.Errorf("error setting status for tms clusters %s: %s", d.Id(), err)
+	}
+
+	if err = d.Set("schedule_id", resp.ScheduleID); err != nil {
+		return diag.Errorf("error setting scheduleId for tms clusters %s: %s", d.Id(), err)
+	}
+
+	if err = d.Set("owner_id", resp.OwnerID); err != nil {
+		return diag.Errorf("error setting ownerId for tms clusters %s: %s", d.Id(), err)
+	}
+
+	if err = d.Set("date_created", resp.DateCreated); err != nil {
+		return diag.Errorf("error setting dateCreated for tms clusters %s: %s", d.Id(), err)
+	}
+
+	if err = d.Set("source", resp.Source); err != nil {
+		return diag.Errorf("error setting source for tms clusters %s: %s", d.Id(), err)
+	}
+
+	if err = d.Set("date_modified", resp.DateModified); err != nil {
+		return diag.Errorf("error setting dateModified for tms clusters %s: %s", d.Id(), err)
+	}
+	if resp.SourceClusters != nil {
+		sourceCls := make([]*string, 0)
+		sourceCls = append(sourceCls, resp.SourceClusters...)
+
+		d.Set("source_clusters", sourceCls)
+	}
+	return nil
+}
+func resourceNutanixNDBTmsClusterUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	updateReq := &era.TmsClusterIntentInput{}
+
+	tmsID := d.Get("time_machine_id")
+	clsID := d.Get("nx_cluster_id")
+	resp, err := conn.Service.ReadTimeMachineCluster(ctx, tmsID.(string), clsID.(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if resp != nil {
+		updateReq.Type = resp.Type
+		updateReq.NxClusterID = resp.NxClusterID
+	}
+
+	if d.HasChange("sla_id") {
+		updateReq.SLAID = utils.StringPtr(d.Get("sla_id").(string))
+		updateReq.ResetSLAID = utils.BoolPtr(true)
+	}
+
+	if d.HasChange("nx_cluster_id") {
+		updateReq.NxClusterID = utils.StringPtr(d.Get("nx_cluster_id").(string))
+	}
+
+	// update Call
+
+	_, er := conn.Service.UpdateTimeMachineCluster(ctx, tmsID.(string), clsID.(string), updateReq)
+	if er != nil {
+		return diag.FromErr(er)
+	}
+	log.Printf("NDB Time Machine Cluster with %s id is updated successfully", d.Id())
+	return resourceNutanixNDBTmsClusterRead(ctx, d, meta)
+}
+func resourceNutanixNDBTmsClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Give a line space in between. 
Is lint not complaining about this?

Reply:
sure. 
No, Lint didn't complain about the spacing.